regparser/notice/xml.py

```diff
@@ -134,6 +134,52 @@ def derive_agencies(self, agencies=None):
         self.xml.insert(0, agencies_el)
         return agency_map
 
+    def derive_cfr_refs(self, refs=None):
+        """
+        Get the references to CFR titles and parts out of the metadata.
+        Return a list of them grouped by title and sorted, for example::
+
+            [{"title": "0", "part": "23"}, {"title": "0", "part": "17"}]
+
+        Becomes::
+
+            [{"title": "0", "parts": ["17", "23"]}]
+
+        Transform the XML to include elements that look like this::
+
+            <EREGS_CFR_REFS>
+                <EREGS_CFR_TITLE_REF title="40">
+                    <EREGS_CFR_PART_REF part="300" />
+                    <EREGS_CFR_PART_REF part="310" />
+                </EREGS_CFR_TITLE_REF>
+            </EREGS_CFR_REFS>
+
+        :arg list refs: The list of title/part pairs.
+        :rtype: list
+        :returns: Grouped & sorted list.
+        """
+        refs = refs if refs else []
+        # Group parts by title:
+        refd = {_["title"]: [] for _ in refs}
+        for ref in refs:
+            refd[ref["title"]].append(ref["part"])
+        refs = [{u"title": k, "parts": refd[k]} for k in refd]
+        # Sort parts and sort list by title:
+        refs = [{u"title": _["title"],
+                 "parts": sorted(_["parts"], key=lambda x: int(x))}
+                for _ in refs]
```

Comment:
It's more succinct, but less clear, so I'm going to leave them as separate steps.